crates/noirc_frontend/src/lexer/token.rs

```diff
@@ -325,7 +325,7 @@ pub enum Attribute {
     Builtin(String),
     Oracle(String),
     Deprecated(Option<String>),
-    Test,
+    Test { expect_failure: bool },
```

Comment:
Will push a commit for an enum variant to make this clearer and more extensible

Reply:
Didn't we want to keep `expect_failure` as a field rather than have there be two separate variants for test functions? I think we should keep the field, although it should be renamed to match the name given to the annotation in the source code: `Test { should_fail: bool }`

Reply:
It would still be one field, its just that should_fail becomes a variant and we call it scope